controllers/bucket_controller.go

```diff
@@ -246,7 +247,9 @@ func (r *BucketReconciler) SetupWithManagerAndOptions(mgr ctrl.Manager, opts Buc
 
 func (r *BucketReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, retErr error) {
 	start := time.Now()
-	log := ctrl.LoggerFrom(ctx)
+	log := ctrl.LoggerFrom(ctx).
+		// Sets a correlation ID to enrich logs from all suboperations.
+		WithValues("cid", uuid.New())
```

Comment:
If we change this to `reconcileID` the field should be compatible with future controller-runtime update.

Reply:
In addition, I think we should aim for using the same constructor for the UUID (PR seems to suggest `uuid.NewUUID()`) which does not result in the value containing hyphens.